specs/capella/light-client/sync-protocol.md

```diff
@@ -0,0 +1,82 @@
+# Capella Light Client -- Sync Protocol
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Constants](#constants)
+- [Containers](#containers)
+  - [Modified `LightClientHeader`](#modified-lightclientheader)
+- [Helper functions](#helper-functions)
+  - [`get_lc_execution_root`](#get_lc_execution_root)
+  - [Modified `is_valid_light_client_header`](#modified-is_valid_light_client_header)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This upgrade adds information about the execution payload to light client data as part of the Capella upgrade. It extends the [Altair Light Client specifications](../../altair/light-client/sync-protocol.md). The [fork document](./fork.md) explains how to upgrade existing Altair based deployments to Capella.
+
+Additional documents describes the impact of the upgrade on certain roles:
+- [Full node](./full-node.md)
+- [Networking](./p2p-interface.md)
+
+## Constants
+
+| Name | Value |
+| - | - |
+| `EXECUTION_PAYLOAD_INDEX` | `get_generalized_index(BeaconBlockBody, 'execution_payload')` (= 25) |
+
+## Containers
+
+### Modified `LightClientHeader`
+
+```python
+class LightClientHeader(Container):
+    # Beacon block header
+    beacon: BeaconBlockHeader
+    # Execution payload header corresponding to `beacon.body_root` (from Capella onward)
+    execution: ExecutionPayloadHeader
```

Comment:
Why did `execution` and `execution_branch` go here rather than in the `LightClientUpdate`? Because `current_sync_committee` and `current_sync_committee_branch`, etc. are located directly on `LightClientUpdate`. I'm wondering if we could have added `execution` and `execution_branch` to `LightClientBootstrap` and `LightClientUpdate` directly rather than introducing `LightClientHeader`.

Reply:
It allows to reuse the entire logic in `sync-protocol.md` across forks (also in the future). Besides the `beacon.state_root` and `beacon.slot`, it doesn't care about any other bundled data except that it is valid. Only the validity criteria needs to be updated across forks (e.g., when a new field is introduced to `ExecutionPayload`).

The `execution_branch` is rooted in `beacon.body_root`, so logically belongs to a particular header (and can be verified with just the info in `LightClientHeader`). For `LightClientUpdate`, you have two of those `execution_branch`es (one for `attested_header`, and one for `finalized_header`).